src/main/java/ua/tc/marketplace/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package ua.tc.marketplace.service.impl;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ua.tc.marketplace.exception.tag.TagInUseException;
+import ua.tc.marketplace.exception.tag.TagNotFoundException;
+import ua.tc.marketplace.exception.user.UserNotFoundException;
+import ua.tc.marketplace.model.dto.tag.CreateTagDto;
+import ua.tc.marketplace.model.dto.tag.TagDto;
+import ua.tc.marketplace.model.dto.tag.UpdateTagDto;
+import ua.tc.marketplace.model.entity.Tag;
+import ua.tc.marketplace.repository.TagRepository;
+import ua.tc.marketplace.service.TagService;
+import ua.tc.marketplace.util.mapper.TagMapper;
+
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of the {@link TagService} interface. Provides methods for creating, retrieving,
+ * updating, and deleting tags.
+ */
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class TagServiceImpl implements TagService {
+
+    private final TagRepository tagRepository;
+    private final TagMapper tagMapper;
+
+    /**
+     * Retrieves a paginated list of all tags.
+     *
+     * @param pageable Pagination information (page number, size, sorting).
+     * @return A page of TagDto objects.
+     */
+    @Override
+    public Page<TagDto> findAll(Pageable pageable) {
+        Page<Tag> tags = tagRepository.findAll(pageable);
+        return tags.stream()
+                .collect(Collectors.collectingAndThen(Collectors.toList(), PageImpl::new))
```

Comment:
Does this new Page has the same `size` and `page` as `pageable`?
Just ensure that it does. If not you can use just `tags.map(tagMapper::toDto);`